packages/node/src/integrations/tracing/redis.ts

```diff
@@ -65,26 +48,22 @@ export const instrumentRedis = generateInstrumentOnce(INTEGRATION_NAME, () => {
         span.setAttributes({ 'network.peer.address': networkPeerAddress, 'network.peer.port': networkPeerPort });
       }
 
+      const cacheOperation = getCacheOperation(redisCommand);
+
+      if (!cacheOperation) return; // redis command unsupported as cache operation
+
       const cacheItemSize = calculateCacheItemSize(response);
       if (cacheItemSize) span.setAttribute(SEMANTIC_ATTRIBUTE_CACHE_ITEM_SIZE, cacheItemSize);
 
-      if (typeof key === 'string') {
-        switch (redisCommand) {
-          case 'get':
-            span.setAttributes({
-              [SEMANTIC_ATTRIBUTE_SENTRY_OP]: 'cache.get_item', // todo: will be changed to cache.get
-              [SEMANTIC_ATTRIBUTE_CACHE_KEY]: key,
-            });
-            if (cacheItemSize !== undefined) span.setAttribute(SEMANTIC_ATTRIBUTE_CACHE_HIT, cacheItemSize > 0);
-            break;
-          case 'set':
-            span.setAttributes({
-              [SEMANTIC_ATTRIBUTE_SENTRY_OP]: 'cache.put',
-              [SEMANTIC_ATTRIBUTE_CACHE_KEY]: key,
-            });
-            break;
-        }
-      }
+      if (cacheOperation === 'cache.get' && cacheItemSize !== undefined)
+        span.setAttribute(SEMANTIC_ATTRIBUTE_CACHE_HIT, cacheItemSize > 0);
+
+      span.setAttributes({
+        [SEMANTIC_ATTRIBUTE_SENTRY_OP]: cacheOperation,
+        [SEMANTIC_ATTRIBUTE_CACHE_KEY]: safeKey,
+      });
+
+      span.updateName(`${safeKey.substring(0, 1024)}...`);
```

Comment:
I think this is not correct, this would always add `...` even if this is not too long, right...? 😅